"YouTube is in full screen" banner is not glass

Phinny01 opened the pull request:
#### d5ba0ada4c94d38cdd42a0f4d438aef1a137b74c
<pre>
&quot;YouTube is in full screen&quot; banner is not glass
<a href="https://bugs.webkit.org/show_bug.cgi?id=299563">https://bugs.webkit.org/show_bug.cgi?id=299563</a>
<a href="https://rdar.apple.com/149900187">rdar://149900187</a>

Reviewed by NOBODY (OOPS!).

Applies _UIViewGlassVariantAVPlayer to the background of the fullscreen dismissal banner - &quot;YouTube
is in fullscreen&quot; to achieve the liquid glass style for the new OS, while maintaining a fallback
for the old OS. Added temporary SPI usage entry for _setBackground: selector used to apply glass
material effects to the fullscreen dismissal banner in Safari.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController showUI]):
(-[WKFullScreenViewController hideUI]):
(-[WKFullScreenViewController configureEnvironmentPickerOrFullscreenVideoButtonView]):
(-[WKFullScreenViewController _removeEnvironmentPickerButtonView]):
(-[WKFullScreenViewController loadView]):
(-[WKFullScreenViewController viewDidLayoutSubviews]):
(-[WKFullScreenViewController _effectiveFullscreenInsets]):
(-[WKFullScreenViewController _applyGlassMaterialToView:]):
(-[WKFullScreenViewController _updateCornerRadiusForStackView:withRadius:]):
(-[WKFullScreenViewController _updateControlsStackConstraints]):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d5ba0ada4c94d38cdd42a0f4d438aef1a137b74c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1927 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40517 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137054 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81132 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/af9ff78f-5da3-497b-9040-56f5b3eac441) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1872 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1816 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98772 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66623 "Passed tests") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/228a89f1-ed3e-42f1-af5b-f57416c3fa95) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132613 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1434 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116130 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79445 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e026ac67-ab3f-467a-9889-fdfd4b1b7147) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1353 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80326 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109826 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34760 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139536 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1721 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1626 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107287 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1766 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112471 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107148 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1392 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30981 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54471 "The change is no longer eligible for processing.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1794 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65160 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1608 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1643 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1715 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
r+ but let's also apply the glass treatment to other controls as we discussed elsewhere!

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,63 @@ - (void)loadView
+        _banner = nil;
+        _bannerLabel = nil;
+        _bannerTapToDismissRecognizer = nil;
+    }
```
I don't think this logic is necessary. `-loadView` should only be called once during the view controller lifecycle.

Phinny01 replied:
You're right. I've removed it

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,63 @@ - (void)loadView
+
+    [[_banner contentView] addSubview:_bannerLabel.get()];
+
+    UIView *content = [_banner contentView];
```
```suggestion
    RetainPtr content = [_banner contentView];
```

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,63 @@ - (void)loadView
     auto banner = adoptNS([[WKFullscreenStackView alloc] init]);
     [banner addArrangedSubview:_bannerLabel.get() applyingMaterialStyle:AVBackgroundViewMaterialStyleSecondary tintEffectStyle:AVBackgroundViewTintEffectStyleSecondary];
     _banner = WTFMove(banner);
 
```
Unnecessary newline here.

aestes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,55 @@ - (void)loadView
+
+    RetainPtr content = [_banner contentView];
+    if (!content)
+        return;
```
`-[UIVisualEffectView contentView]` is a nonnull property, so it can never return nil. Let's remove the null check. Also, `contentView` is a better name than `content`.

aestes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,55 @@ - (void)loadView
+    if (!content)
+        return;
+
+    RetainPtr constraints = adoptNS([[NSMutableArray alloc] init]);
```
This isn't needed anymore.

aestes commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
```diff
@@ -822,18 +826,55 @@ - (void)loadView
+        [[_bannerLabel bottomAnchor] constraintEqualToAnchor:[content bottomAnchor] constant:-bannerLabelInset],
+    ]];
+
+    [NSLayoutConstraint activateConstraints:constraints.get()];
```
This isn't needed anymore.

Commit:
"YouTube is in full screen" banner is not glass
https://bugs.webkit.org/show_bug.cgi?id=299563
rdar://149900187

Reviewed by NOBODY (OOPS!).

Applies _UIViewGlassVariantAVPlayer to the background of the fullscreen dismissal banner - "YouTube
is in fullscreen" to achieve the liquid glass style for the new OS, while maintaining a fallback
for the old OS. Added temporary SPI usage entry for _setBackground: selector used to apply glass
material effects to the fullscreen dismissal banner in Safari.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController showUI]):
(-[WKFullScreenViewController hideUI]):
(-[WKFullScreenViewController configureEnvironmentPickerOrFullscreenVideoButtonView]):
(-[WKFullScreenViewController _removeEnvironmentPickerButtonView]):
(-[WKFullScreenViewController loadView]):
(-[WKFullScreenViewController viewDidLayoutSubviews]):
(-[WKFullScreenViewController _effectiveFullscreenInsets]):
(-[WKFullScreenViewController _applyGlassMaterialToView:]):
(-[WKFullScreenViewController _updateCornerRadiusForStackView:withRadius:]):
(-[WKFullScreenViewController _updateControlsStackConstraints]):
